pkg/diagnose/runner.go

```diff
@@ -376,54 +377,38 @@ func RunInAgentProcess(diagCfg diagnosis.Config, deps SuitesDepsInAgentProcess)
 	return getDiagnosesFromCurrentProcess(diagCfg, suites)
 }
 
-// RunStdOutInAgentProcess enumerates registered Diagnose suites and get their diagnoses
-// for human consumption
-func RunStdOutInAgentProcess(w io.Writer, diagCfg diagnosis.Config, deps SuitesDepsInAgentProcess) error {
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		return RunInAgentProcess(diagCfg, deps)
-	})
+// RunLocalCheck runs locally the checks created by the registries.
+func RunLocalCheck(diagCfg diagnosis.Config, registries ...func(*diagnosis.Catalog)) ([]diagnosis.Diagnoses, error) {
+	suites := buildCustomSuites(registries...)
+	return getDiagnosesFromCurrentProcess(diagCfg, suites)
 }
 
-// RunStdOutInCLIProcess enumerates registered Diagnose suites and get their diagnoses
-// for human consumption
-func RunStdOutInCLIProcess(w io.Writer, diagCfg diagnosis.Config, deps SuitesDepsInCLIProcess) error {
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		return RunInCLIProcess(diagCfg, deps)
-	})
+// RunDiagnoseStdOut runs the diagnose and outputs the results to the writer.
+func RunDiagnoseStdOut(w io.Writer, diagCfg diagnosis.Config, diagnoses []diagnosis.Diagnoses) error {
+	if diagCfg.JSONOutput {
+		return runStdOutJSON(w, diagnoses)
+	}
+	return runStdOut(w, diagCfg, diagnoses)
 }
 
-// RunStdOutLocalCheck runs locally the checks created by the registries.
-func RunStdOutLocalCheck(w io.Writer, verbose bool, registries ...func(*diagnosis.Catalog)) error {
-	diagCfg := diagnosis.Config{Verbose: verbose, RunLocal: true}
-	return runStdOut(w, diagCfg, func(diagCfg diagnosis.Config) ([]diagnosis.Diagnoses, error) {
-		suites := buildCustomSuites(registries...)
+func runStdOutJSON(w io.Writer, diagnoses []diagnosis.Diagnoses) error {
+	diagJSON, err := json.MarshalIndent(diagnoses, "", "  ")
+	if err != nil {
+		fmt.Fprintf(w, (fmt.Sprintf("{error: marshalling diagnose results to JSON: %s}\n", err)))
```

Comment:
Nit: Should we use a map instead of JSON directly ourselves? 

```go
body, _ := json.Marshal(map[string]string{"error": fmt.Sprintf("marshalling diagnose results to JSON: %s", err)})
```

Reply:
good catch thx